src/librustc_save_analysis/dump_visitor.rs

```diff
@@ -1297,7 +1298,14 @@ impl<'v, 'l, 'tcx: 'l, 'll, D: Dump +'ll> Visitor<'v> for DumpVisitor<'l, 'tcx,
             ast::ExprKind::TupField(ref sub_ex, idx) => {
                 self.visit_expr(&sub_ex);
 
-                let hir_node = self.save_ctxt.tcx.map.expect_expr(sub_ex.id);
+                let hir_node = match self.save_ctxt.tcx.map.find(sub_ex.id) {
+                    Some(Node::NodeExpr(expr)) => expr,
+                    _ => {
+                        debug!("Missing or weird node for sub-expression {} in {:?}",
+                               sub_ex.id, ex);
```

Comment:
My reasoning is that if subex is messed up for some other reason, then it is useful to have an expression which we know is not messed up. I also find it easier to debug this way - you get a reference to say `(a, b, x + 4)` rather than `a` and the former is easier to grep for.